options/posix/generic/netdb-stubs.cpp

```diff
@@ -125,9 +125,78 @@ struct netent *getnetent(void) {
 	__builtin_unreachable();
 }
 
-struct hostent *gethostbyname(const char *) {
-	__ensure(!"gethostbyname() not implemented");
-	__builtin_unreachable();
+struct hostent *gethostbyname(const char *name) {
+	if (!name) {
+		h_errno = HOST_NOT_FOUND;
+		return NULL;
+	}
+
+	struct mlibc::dns_addr_buf buf[NAME_MAX];
```

Comment:
This looks strange: it's a large array of moderately-sized structs. Doesn't this require huge amounts of (stack) space?